src/app/auth/profiles/profiles.component.ts

```diff
@@ -0,0 +1,131 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { AppConfig } from '../../../environments/environment';
+import { MatDialog } from '@angular/material';
+import {
+  JsonParseErrorDialogComponent
+} from './json-parse-error-dialog/json-parse-error-dialog.component';
+import ProfilesJson from './profiles.json';
+
+/**
+ * Indicates all the elements that make up a Profile.
+ */
+export interface Profile {
+  profileName: string;
+  username: string;
+  password: string;
+  encodedText: string;
+}
+
+/**
+ * This component is responsible allowing the user to select a Profile
+ * in the authentication page.
+ */
+
+@Component({
+  selector: 'app-profiles',
+  templateUrl: './profiles.component.html',
+  styleUrls: ['./profiles.component.css']
+})
+export class ProfilesComponent implements OnInit {
+
+  profiles: Profile[] = undefined; // List of profiles taken from profiles.json
+  blankProfile: Profile = {profileName: '', password: '', username: '',
+    encodedText: ''}; // A blank profile to reset values
```

Comment:
Isn't the indentation for line wrapping double the normal amount?

Reply:
Yes, according to the [google's javascript coding style](https://google.github.io/styleguide/jsguide.html#formatting-indent),

> When line-wrapping, each line after the first (each continuation line) is indented at least +4 from the original line, unless it falls under the rules of block indentation.
>
> When there are multiple continuation lines, indentation may be varied beyond +4 as appropriate. In general, continuation lines at a deeper syntactic level are indented by larger multiples of 4, and two lines use the same indentation level if and only if they begin with syntactically parallel elements.

I think this is due to IntelliJ reading the coding style used in the files of the project (Using a norm of 2 spaces for an Indent) and performing a generic application of this on all indents. I guess the only way for this is manually double-tabbing or preventing IntelliJ from automatically reading the code style and applying a fixed style over the project files.

Reply:
@JunWei96 @RonakLakhotia any idea on what action to take?

Reply:
I think the settings can be adjusted under `Code Style` in IntelliJ.  @ptvrajsk try looking up some instructions online. For instance see [this](https://stackoverflow.com/questions/28426355/intellij-code-style-setting-for-wrapping-on-multi-line-function-arguments).
Besides is there a need to break and move to a new line? If it is not violating the coding standards of the project I think it would be better if you complete the code in the original line.

Reply:
> I think the settings can be adjusted under `Code Style` in IntelliJ. @ptvrajsk try looking up some instructions online. For instance see [this](https://stackoverflow.com/questions/28426355/intellij-code-style-setting-for-wrapping-on-multi-line-function-arguments).
> Besides is there a need to break and move to a new line? If it is not violating the coding standards of the project I think it would be better if you complete the code in the original line.

Ah alright, I was previously stick-ing to the 80 column limit in the [js coding style](https://google.github.io/styleguide/jsguide.html#formatting-column-limit). I'll re-extend the multi-lined code.